source/classes.tex

```diff
@@ -2260,10 +2260,9 @@
 \pnum
 \indextext{destructor!virtual}%
 \indextext{destructor!pure virtual}%
-A prospective destructor can be declared
-\tcode{virtual}\iref{class.virtual}
-or pure
-\tcode{virtual}\iref{class.abstract}.
+A prospective destructor can be
+a virtual\iref{class.virtual}
```

Comment:
We're slightly changing the semantics here. Previously, we said we can declare those things with the virtual keyword, now we're saying they're virtual functions (which might be endowed due to inheritance).

Maybe: "A prospective destructor can be declared `virtual` (class.virtual), including as a pure virtual function (class.abstract)."

Reply:
I don't think we are changing semantics here -- prospective destructors are still non-static member functions, thus [class.virtual] p1 applies (they can be virtual either though the use of the specifier, or through overriding), and so does [class.virtual] p2 (a destructor can override one declared in a base class). This paragraph and [class.virtual[ p7 reinforce that virtual prospective destructors are allowed, but aren't what permit this behavior.

Reply:
@jensmaurer (while looking at [class.virtual] p2 it looks like ref-qualifier should be a grammar term and "cv-qualification" should be _cv-qualifier-seq_. I'll open a seperate PR)

Reply:
This sentence could probably just be turned into a note in #3966

Reply:
I think a prospective destructor never overrides anything (it's still a templated entity). Only at instantiation time do you choose the destructor, and that might be a virtual function via inheritance.

Reply:
A prospective destructor doesn't necessarily have to be a templated entity:
```
struct T
{
  virtual ~T(); // prospective destructor
}; // selected destructor
```

Similarly, to permit the use of `override`, a prospective destructor would _need_ to be a virtual function by inheritance.

Reply:
A virtual functions can't have a trailing-requires-clause [class.virtual] p6, so my point is moot, I believe.

Reply:
I've thought about this some more. I think we should clearly distinguish between "declared `virtual`" and "virtual function". The former refers to the token appearing in the class definition, the latter refers to the resulting attribute of the member function. The original sentence referred to the token (make that `\keyword`), so we should retain that.
In order not to mix the levels here, maybe saying "A prospective destructor can be declared `virtual` (class.virtual) and with a _pure-specifier_ (class.abstract)." limts the statement to the syntax. The rest follows from the referenced places.

Reply:
Limiting it to syntax is reasonable -- perhaps we add a note saying that it may be virtual by inheritance?